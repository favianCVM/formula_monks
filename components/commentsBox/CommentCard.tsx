import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Text} from '../fragments/Text';
import {PostComment} from '../../views/PostDetails';
import {COLORS} from '../../styles/colors';

interface CommentCardProps extends PostComment {}

const styles = StyleSheet.create({
  container: {
    borderColor: COLORS.text,
    borderWidth: 1,
    borderRadius: 8,
    marginBottom: 10,
    padding: 8,
  },
  separator: {
    backgroundColor: COLORS.text,
    height: 1,
    marginVertical: 4,
  },
});

const CommentCard = React.memo(({body, email, id, name}: CommentCardProps) => {
  return (
    <View style={styles.container}>
      <Text>{email}</Text>
      <View style={styles.separator}></View>
      <Text>{body}</Text>
    </View>
  );
});

export {CommentCard};
