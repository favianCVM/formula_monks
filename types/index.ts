export interface Post {
  userId: number;
  id: number;
  title: string;
  body: string;
}

export interface Rating {
  id: Post['id'];
  value: number;
}

export interface PostComment {
  postId: number;
  id: number;
  name: string;
  email: string;
  body: string;
}

export interface AuthorInformation {
  id: number;
  name: string;
  username: string;
  email: string;
  address: {
    street: string;
    city: string;
  };
  phone: string;
  website: string;
  company: {
    name: string;
  };
}

export interface PostDetails {
  id: Post['id'];
  author: AuthorInformation;
  comments: Array<PostComment>;
}
