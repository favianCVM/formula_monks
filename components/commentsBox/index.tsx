import React from 'react';
import {PostComment} from '../../views/PostDetails';
import {CommentCard} from './CommentCard';
import {FlatList, GestureHandlerRootView} from 'react-native-gesture-handler';
import {Dimensions} from 'react-native';
import {normalize} from '../../libs/normalizeSize';
import {Title} from '../fragments/Title';

interface CommentsBoxProps {
  comments: Array<PostComment>;
}

const CommentsBox = React.memo(({comments}: CommentsBoxProps) => {
  return (
    <GestureHandlerRootView style={{flex: 1}}>
      <FlatList
        contentContainerStyle={{
          paddingHorizontal: normalize(15),
        }}
        data={comments}
        keyExtractor={({id}) => `post-comment-${id}`}
        renderItem={({item}) => <CommentCard {...item} />}
      />
    </GestureHandlerRootView>
  );
});

export {CommentsBox};
