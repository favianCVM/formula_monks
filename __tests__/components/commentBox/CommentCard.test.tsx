import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {CommentCard} from '../../../components/commentsBox/CommentCard';
import {PostComment} from '../../../types';
import renderer from 'react-test-renderer';

describe('Comment card element', () => {
  const commentPlaceholder: PostComment = {
    id: 1,
    postId: 1,
    body: 'lorem ipsum dolor sit amet',
    email: 'loremipsum dolor sit amet',
    name: 'loremipsum dolor sit amet',
  };

  it('renders comment card with the given values', () => {
    const {getByTestId} = render(<CommentCard {...commentPlaceholder} />);

    const commentCardElement = getByTestId('CommentCardElement');
    expect(commentCardElement).not.toBeNull();

    expect(screen.getByText(commentPlaceholder.body)).toBeVisible();
    expect(screen.getByText(commentPlaceholder.email)).toBeVisible();
  });

  it('renders comment card with empty values', () => {
    const {queryByTestId} = render(<CommentCard />);

    const commentBody = queryByTestId('CommentCardBody');
    expect(commentBody).toHaveTextContent('');

    const commentEmail = queryByTestId('CommentCardEmail');
    expect(commentEmail).toHaveTextContent('');
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<CommentCard {...commentPlaceholder} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
