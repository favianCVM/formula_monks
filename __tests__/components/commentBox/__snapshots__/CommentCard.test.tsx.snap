// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment card element renders correctly 1`] = `
<View
  style={
    {
      "borderColor": "#1B1C1E",
      "borderRadius": 8,
      "borderWidth": 1,
      "marginBottom": 10,
      "padding": 8,
    }
  }
  testID="CommentCardElement"
>
  <Text
    style={
      [
        {
          "color": "#1B1C1E",
          "fontSize": 16,
        },
        undefined,
      ]
    }
    testID="CommentCardEmail"
  >
    loremipsum dolor sit amet
  </Text>
  <View
    style={
      {
        "backgroundColor": "#1B1C1E",
        "height": 1,
        "marginVertical": 4,
      }
    }
  />
  <Text
    style={
      [
        {
          "color": "#1B1C1E",
          "fontSize": 16,
        },
        undefined,
      ]
    }
    testID="CommentCardBody"
  >
    lorem ipsum dolor sit amet
  </Text>
</View>
`;
